### PR TITLE
EVG-5943: merge task

### PR DESCRIPTION
**Pull request by ybrill**
Make a task from the previously created "git.push" command and add it to the patch version created to test a CLI commit queue item. The task will run once all other tasks in the patch have completed successfully.
Separately, add configuration for the Commit Queue merge to the admin page.

### CODE REVIEW

File: service/templates/admin.html [hunk 1]
```diff
@@ -551,7 +551,19 @@ <h2 class="modal-title">[[modalTitle]]</h2>
...
 											</md-card-title>
 											<md-card-content>
-												<button ng-click="clearCommitQueues()" class="btn btn-primary">Clear Commit Queues</button>
+												<md-input-container class="control" style="width:45%;">
+														<label>Merge Distro</label>
+														<input type="text" ng-model="Settings.commit_queue.merge_task_distro">
+												</md-input-container>
+												<md-input-container class="control" style="width:45%;">
+													<label>Committer Name</label>
+													<input type="text" ng-model="Settings.Settings.commit_queue.committer_name">
```
Comment (john-m-liu) [inline service/templates/admin.html]: Settings is repeated

File: units/commit_queue.go [hunk 2]
```diff
@@ -476,3 +493,65 @@ func validateBranch(branch *github.Branch) error {
...
+				Name:    "*",
+				Variant: "*",
+			},
+		},
+	}
+
+	project.BuildVariants = append(project.BuildVariants, mergeBuildVariant)
+	project.Tasks = append(project.Tasks, mergeTask)
+
+	validationErrors := validator.CheckProjectSemantics(project)
```
Comment (john-m-liu) [inline units/commit_queue.go]: What is it that we need to validate here that hasn't been done already?

File: service/templates/admin.html (hunk 1 above)
Comment (ybrill) [reply service/templates/admin.html]: fixed 😌

File: units/commit_queue.go (hunk 2 above)
Comment (ybrill) [reply units/commit_queue.go]: Meant to do CheckProjectSyntax (updated now) to run `validateProjectTaskNames` and `validateBVNames` to check the project didn't already have a task/bv with the same name.

**Comment by ybrill** [comment]
If the merge was already successful the new attempt to merge will fail since HEAD has changed since the item reached the head of the queue: 
https://github.com/evergreen-ci/evergreen/blob/927f87971737a418741caecc9b2955d54aa1c63e/command/git_push.go#L70
If the merge failed the merge might succeed if no other merges happened in the meantime. If it does, the next merge will fail if it's already been taken off the queue.

File: units/commit_queue.go (hunk 2 above)
Comment (ybrill) [reply units/commit_queue.go]: It also makes sure the distro configured for the task to run on actually exists.

**Pull request merged**